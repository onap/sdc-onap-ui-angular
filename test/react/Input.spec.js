import React from 'react';
import Input from '../../src/react/Input.js';

import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

class InputForm extends React.Component {
	constructor(props) {
		super(props);
		this.state = {value: 'Initial'};
		this.handleChange = this.handleChange.bind(this);
	}

	handleChange(val) {
		this.setState({value: val});
	}

	getValue() {
		return this.input.getValue();
	}

	render() {
		return (
			<form >
				<Input ref={(input)=>{this.input = input;}} value={this.state.value} name='testinput' onChange={this.handleChange} />
			</form>
		);
	}
}

describe('Input', () => {
	test('Input - default', () => {
		const input = renderer.create(<Input label='label for input' name='clickme' type='text'/>).toJSON();
		expect(input).toMatchSnapshot();
	});

	test('Input - required', () => {
		const input = renderer.create(<Input required label='label for input' name='clickme' type='text'/>).toJSON();
		expect(input).toMatchSnapshot();
	});

	test('Input - number', () => {
		const input = renderer.create(<Input  label='label for input' name='clickme' type='number'/>).toJSON();
		expect(input).toMatchSnapshot();
	});

	test('Input - readonly', () => {
		const input = renderer.create(<Input  label='label for input' name='clickme' type='text' readOnly/>).toJSON();
		expect(input).toMatchSnapshot();
	});

	test('Input - placeholder', () => {
		const input = renderer.create(<Input  label='label for input' name='clickme' type='text' placeholder='hint'/>).toJSON();
		expect(input).toMatchSnapshot();
	});

	test('Input - error', () => {
		const input = renderer.create(<Input  label='label for input' name='clickme' type='text' errorMessage='this is an error'/>).toJSON();
		expect(input).toMatchSnapshot();
	});

	test('Input - checked state value changes', () => {
		const input = mount(<InputForm />);
		expect(input.instance().getValue()).toEqual('Initial');
		input.find('input').simulate('change', { target : { value: 'Changed' }});
		expect(input.instance().getValue()).toEqual('Changed');
		expect(input.find('input').prop('value')).toEqual('Changed');
	});
});
