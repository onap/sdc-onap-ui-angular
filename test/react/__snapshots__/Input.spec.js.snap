// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Input - default 1`] = `
<div
  className="sdc-input "
>
  <label
    className="sdc-input__label  "
    htmlFor="clickme"
  >
    label for input
  </label>
  <input
    className="sdc-input__input  "
    data-test-id={undefined}
    disabled={false}
    id="clickme"
    name="clickme"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder={undefined}
    readOnly={false}
    type="text"
    value={undefined}
  />
</div>
`;

exports[`Input Input - error 1`] = `
<div
  className="sdc-input "
>
  <label
    className="sdc-input__label  "
    htmlFor="clickme"
  >
    label for input
  </label>
  <input
    className="sdc-input__input error "
    data-test-id={undefined}
    disabled={false}
    id="clickme"
    name="clickme"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder={undefined}
    readOnly={false}
    type="text"
    value={undefined}
  />
  <div
    className="sdc-label__error"
  >
    <div
      className="svg-icon-wrapper   __negative  right"
      disabled={undefined}
      onClick={undefined}
    >
      <svg />
      <span
        className="svg-icon-label "
      >
        this is an error
      </span>
    </div>
  </div>
</div>
`;

exports[`Input Input - number 1`] = `
<div
  className="sdc-input "
>
  <label
    className="sdc-input__label  "
    htmlFor="clickme"
  >
    label for input
  </label>
  <input
    className="sdc-input__input  "
    data-test-id={undefined}
    disabled={false}
    id="clickme"
    name="clickme"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder={undefined}
    readOnly={false}
    type="number"
    value={undefined}
  />
</div>
`;

exports[`Input Input - placeholder 1`] = `
<div
  className="sdc-input "
>
  <label
    className="sdc-input__label  "
    htmlFor="clickme"
  >
    label for input
  </label>
  <input
    className="sdc-input__input  "
    data-test-id={undefined}
    disabled={false}
    id="clickme"
    name="clickme"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="hint"
    readOnly={false}
    type="text"
    value={undefined}
  />
</div>
`;

exports[`Input Input - readonly 1`] = `
<div
  className="sdc-input "
>
  <label
    className="sdc-input__label view-only "
    htmlFor="clickme"
  >
    label for input
  </label>
  <input
    className="sdc-input__input  view-only"
    data-test-id={undefined}
    disabled={false}
    id="clickme"
    name="clickme"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder={undefined}
    readOnly={true}
    type="text"
    value={undefined}
  />
</div>
`;

exports[`Input Input - required 1`] = `
<div
  className="sdc-input "
>
  <label
    className="sdc-input__label  "
    htmlFor="clickme"
  >
    label for input
  </label>
  <input
    className="sdc-input__input  "
    data-test-id={undefined}
    disabled={false}
    id="clickme"
    name="clickme"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder={undefined}
    readOnly={false}
    type="text"
    value={undefined}
  />
</div>
`;
